plugins/dynamix/include/FileTree.php

```diff
@@ -24,70 +25,38 @@
+$match   = ($_POST['match'] ?? '.*');
 
 // set checkbox if multiSelect set to true
-$checkbox = ( isset($_POST['multiSelect']) && $_POST['multiSelect'] == 'true' ) ? "<input type='checkbox' />" : null;
-
-$returnDir	= $postDir;
+$checkbox = (isset($_POST['multiSelect']) && $_POST['multiSelect']=='true') ? "<input type='checkbox'>" : "";
 
 echo "<ul class='jqueryFileTree'>";
-
 // Parent dirs
-if ($_POST['show_parent'] == "true" ) {
-	echo "<li class='directory collapsed'>{$checkbox}<a href='#' rel='" . htmlspecialchars(dirname($postDir), ENT_QUOTES) . "/'>..</a></li>";
-}
-
-if( file_exists($postDir) ) {
-
-	$files = scandir($postDir);
-
-	natcasesort($files);
-
-	if( count($files) > 2 ) { // The 2 accounts for . and ..
-
-		foreach( $files as $file ) {
-			if( file_exists($postDir . $file) && $file != '.' && $file != '..' ) {
-				if( is_dir($postDir . $file) ) {
-					$htmlRel	= htmlspecialchars($returnDir . $file, ENT_QUOTES);
-					$htmlName	= htmlspecialchars((strlen($file) > 33) ? substr($file,0,33).'...' : $file);
-
-					echo "<li class='directory collapsed'>{$checkbox}<a href='#' rel='" . $htmlRel . "/'>" . $htmlName . "</a></li>";
-				}
-			}
-		}
-
-		// All files
-		foreach( $files as $file ) {
-			if( file_exists($postDir . $file) && $file != '.' && $file != '..' ) {
-				if( !is_dir($postDir . $file) ) {
-					$htmlRel	= htmlspecialchars($returnDir . $file, ENT_QUOTES);
-					$htmlName	= htmlspecialchars($file);
-					$ext		= strtolower(preg_replace('/^.*\./', '', $file));
-
-    				foreach ($filters as $filter) {
-						if (empty($filter) | $ext==$filter) {
-							if (empty($match) || preg_match('/'.$match.'/', $file)) {
-								echo "<li class='file ext_{$ext}'>{$checkbox}<a href='#' rel='" . $htmlRel . "'>" . $htmlName . "</a></li>";
-							}
-						}
-					}
-				}
-			}
-		}
-
-	}
+if ($_POST['show_parent']=='true') echo "<li class='directory collapsed'>$checkbox<a href='#' rel=\"".htmlspecialchars(dirname($rootdir))."/\">..</a></li>";
+
+if (is_dir($rootdir)) {
+  if (mb_substr($rootdir,-1)!='/') $rootdir .= '/';
+  $names = array_filter(scandir($rootdir),function($n){return $n!='.' && $n!='..';});
+  if (count($names)) {
+    natcasesort($names);
+    foreach ($names as $dir) if (is_dir($rootdir.$dir)) {
+      $htmlRel  = htmlspecialchars($rootdir.$dir);
+      $htmlName = htmlspecialchars(mb_strlen($dir)<=33 ? $dir : mb_substr($dir,0,33).'...');
+      echo "<li class='directory collapsed'>$checkbox<a href='#' rel=\"$htmlRel/\">$htmlName</a></li>";
+    }
+    foreach ($names as $file) if (is_file($rootdir.$file)) {
+      $htmlRel  = htmlspecialchars($rootdir.$file);
+      $htmlName = htmlspecialchars($file);
+      $ext      = mb_strtolower(pathinfo($file)['extension']);
+      foreach ($filters as $filter) if (empty($filter)||$ext==$filter) {
+        if (empty($match)||preg_match("#$match#",$file)) echo "<li class='file ext_$ext'>$checkbox<a href='#' rel=\"$htmlRel\">$htmlName</a></li>";
```

Comment:
$_POST should be sanitised to prevent injecting into preg_match to cause a ReDos type attack.